From 9af185be6068d279736c0d2b2162a4e768444d3d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 29 May 2021 00:41:32 -0700
Subject: [PATCH] gsec: Fix build on musl

musl defines iovec struct in alltypes.h file which is included by many
system include files, therefore check for appropriate define from musl
before defining it again

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/core/tsi/alts/crypt/gsec.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/core/tsi/alts/crypt/gsec.h b/src/core/tsi/alts/crypt/gsec.h
index 4d65caa944..3c367b2895 100644
--- a/src/core/tsi/alts/crypt/gsec.h
+++ b/src/core/tsi/alts/crypt/gsec.h
@@ -26,11 +26,14 @@
 #include <stdlib.h>
 
 #include <grpc/grpc.h>
+#include <sys/uio.h>
 
+#if !defined(__DEFINED_struct_iovec)
 struct iovec {
   void* iov_base;
   size_t iov_len;
 };
+#endif
 
 /**
  * A gsec interface for AEAD encryption schemes. The API is thread-compatible.
-- 
2.31.1

